x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/querydsl/query/SingleValueQuery.java

```diff
@@ -107,7 +107,15 @@ public static class Builder extends AbstractQueryBuilder<Builder> {
             this.next = in.readNamedWriteable(QueryBuilder.class);
             this.field = in.readString();
             if (in.getTransportVersion().onOrAfter(TransportVersions.ESQL_SINGLE_VALUE_QUERY_SOURCE)) {
-                this.source = Source.readFrom((PlanStreamInput) in);
+                if (in instanceof PlanStreamInput psi) {
+                    this.source = Source.readFrom(psi);
+                } else {
+                    /*
+                     * For things like CanMatch we serialize without the Source. But we
```

Comment:
I couldn't find a `CanMatch.java`. Could we maybe rephrase the comment here a little so it's clear when this would be deserialized without a `PlanStreamInput`? It's currently unclear to me.